src/DetailsView/actions/tab-stop-requirement-action-message-creator.ts

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+import {
+    UpdateTabStopRequirementStatusPayload,
+    AddTabStopInstancePayload,
+    UpdateTabStopInstancePayload,
+    RemoveTabStopInstancePayload,
+} from 'background/actions/action-payloads';
+import { DevToolActionMessageCreator } from 'common/message-creators/dev-tool-action-message-creator';
+import { Messages } from 'common/messages';
+import { TabStopRequirementId } from 'types/tab-stop-requirement-info';
+import { TabStopRequirementStatus } from '../../common/types/store-data/visualization-scan-result-data';
+const messages = Messages.Visualizations.TabStops;
+
+export class TabStopRequirementActionMessageCreator extends DevToolActionMessageCreator {
+    public addTabStopInstance(requirementId: TabStopRequirementId, description: string): void {
+        const telemetry = this.telemetryFactory.forTabStopRequirement(requirementId);
+
+        const payload: AddTabStopInstancePayload = {
+            requirementId,
+            description,
+            telemetry,
+        };
+
+        this.dispatcher.dispatchMessage({
+            messageType: messages.AddTabStopInstance,
+            payload,
+        });
+    }
+
+    public updateTabStopInstance(
+        requirementId: TabStopRequirementId,
+        id: string,
+        description: string,
+    ): void {
+        const telemetry = this.telemetryFactory.forTabStopRequirement(requirementId);
+
+        const payload: UpdateTabStopInstancePayload = {
+            requirementId,
+            description,
+            id,
+            telemetry,
+        };
+
+        this.dispatcher.dispatchMessage({
+            messageType: messages.UpdateTabStopInstance,
+            payload,
+        });
+    }
+
+    public removeTabStopInstance(requirementId: TabStopRequirementId, id: string): void {
+        const telemetry = this.telemetryFactory.forTabStopRequirement(requirementId);
+        const payload: RemoveTabStopInstancePayload = {
+            requirementId,
+            id,
+            telemetry,
+        };
+
+        this.dispatcher.dispatchMessage({
+            messageType: messages.RemoveTabStopInstance,
+            payload,
+        });
+    }
+
+    public updateTabStopRequirementStatus(
+        requirementId: TabStopRequirementId,
+        status: TabStopRequirementStatus,
+    ): void {
+        const telemetry = this.telemetryFactory.forTabStopRequirement(requirementId);
```

Comment:
We usually send telemetry for how the action was initiated/triggered (keyboard or mouse for example); maybe we should do that here (or check with PM team on whether we still need to do that).

Reply:
I based it on the instances actions from DetailsView but would like to know what is preferred

Reply:
Yeah, on a look-through, it's fairly inconsistent. Let's check with the PM team on what they would like.

Reply:
Just checked with PM team on this: this is good as is!